independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java

```diff
@@ -52,54 +59,87 @@ public void handle(Throwable event) {
         future.onSuccess(new Handler<HttpClientRequest>() {
             @Override
             public void handle(HttpClientRequest httpClientRequest) {
-                Buffer actualEntity;
-                try {
-                    actualEntity = ClientSendRequestHandler.this
-                            .setRequestHeadersAndPrepareBody(httpClientRequest, requestContext);
-                } catch (Throwable e) {
-                    requestContext.resume(e);
-                    return;
-                }
-
                 Future<HttpClientResponse> sent;
-                if (actualEntity == AsyncInvokerImpl.EMPTY_BUFFER) {
-                    sent = httpClientRequest.send();
+                if (requestContext.isMultipart()) {
+                    Promise<HttpClientRequest> requestPromise = Promise.promise();
+                    MultipartFormUpload actualEntity;
+                    try {
+                        actualEntity = ClientSendRequestHandler.this.setMultipartHeadersAndPrepareBody(httpClientRequest,
+                                requestContext);
+
+                        Pipe<Buffer> pipe = actualEntity.pipe(); // Shouldn't this be called in an earlier phase ?
```

Comment:
Should it? Did you copy this code from somewhere else?

Reply:
I don't know, this code is copied from vertx-web, with the comment.

Reply:
Okay, that's all I needed 😁